packages/web/src/components/gcds-link/gcds-link.tsx

```diff
@@ -0,0 +1,220 @@
+  shadow: true,
+})
+export class GcdsLink {
+  @Element() el: HTMLElement;
+
+  private shadowElement?: HTMLElement;
+
+  /**
+   * Props
+   */
+
+  /**
+   * Set the link size
+   */
+  @Prop({ mutable: true }) size: 'regular' | 'small' = 'regular';
+
+  @Watch('size')
+  validateSize(newValue: string) {
+    const values = ['regular', 'small'];
+
+    if (!values.includes(newValue)) {
+      this.size = 'regular';
+    }
+  }
+
+  // TODO: Do we need to validate display, rel, target, etc?
+
+  private handleClick = (e: Event) => {
+    if (this.clickHandler) {
+      this.clickHandler(e);
+    }
+  };
+
+  /**
+   * Link props
+   */
+
+  /**
+   * Sets the display behavior of the link
+   */
+  @Prop() display: 'block' | 'inline' | 'inline-block' = 'inline';
+
+  /**
+   * The href attribute specifies the URL of the page the link goes to
+   */
+  @Prop() href = '#';
+
+  /**
+   * The rel attribute specifies the relationship between the current document and the linked document
+   */
+  @Prop() rel: string | undefined;
+
+  /**
+   * The target attribute specifies where to open the linked document
+   */
+  @Prop() target: string | undefined;
+
+  /**
+   * Whether the link is external or not
+   */
+  @Prop() external: false;
+
+  /**
+   * The download attribute specifies that the target (the file specified in the href attribute) will be downloaded when a user clicks on the hyperlink
+   */
+  @Prop() download: string | undefined;
+
+  /**
+   * The type specifies the media type of the linked document
+   */
+  @Prop() type: string | undefined;
+
+  /**
+   * Custom callback function on click event
+   */
+  @Prop() clickHandler: Function;
+
+  /**
+   * Set additional HTML attributes not available in component properties
+   */
+  @State() inheritedAttributes: Object = {};
+
+  /**
+   * Language of rendered component
+   */
+  @State() lang: string;
+
+  /**
+   * Events
+   */
+
+  /**
+   * Emitted when the link has focus.
+   */
+  @Event() gcdsFocus!: EventEmitter<void>;
+
+  /**
+   * Emitted when the link loses focus.
+   */
+  @Event() gcdsBlur!: EventEmitter<void>;
+
+  /*
+   * Observe lang attribute change
+   */
+  updateLang() {
+    const observer = new MutationObserver(mutations => {
+      if (mutations[0].oldValue != this.el.lang) {
+        this.lang = this.el.lang;
+      }
+    });
+    observer.observe(this.el, observerConfig);
+  }
+
+  componentWillLoad() {
+    // Validate attributes and set defaults
+    this.validateSize(this.size);
+
+    this.inheritedAttributes = inheritAttributes(this.el, this.shadowElement, [
+      'referrerpolicy',
+    ]);
+
+    // Define lang attribute
+    this.lang = assignLanguage(this.el);
+
+    this.updateLang();
+  }
+
+  render() {
+    const {
+      size,
+      lang,
+      display,
+      href,
+      rel,
+      target,
+      external,
+      download,
+      type,
+      inheritedAttributes,
+    } = this;
+
+    const Tag = 'a';
+    const attrs = {
+      display,
```

Comment:
`display` isn't a HTML attribute so it won't apply the desired display to the attribute.

You will need to create a class to apply the desired `display` property to the `:host` element. An example of this can be found in the new Text component

Reply:
For future reference to myself, found the code here: https://github.com/cds-snc/gcds-components/blob/ac5d6e7db5b0ec3155441ff33d243c27d6322854/packages/web/src/components/gcds-text/gcds-text.tsx#L184